src/flatdirectory.js

```diff
@@ -210,7 +213,7 @@ export class FlatDirectory {
         let gasLimit = 0;
         const [cost, oldChunkLength, maxFeePerBlobGas, gasFeeData] = await Promise.all([
             fileContract.upfrontPayment(),
-            fileContract.countChunks(hexName),
+            this.#countChunks(fileContract, hexName),
```

Comment:
There are similar code between estimateCost && upload (for both calldata and blob), can we have a internal function for the similar code?

Reply:
The content obtained concurrently by `upload` is different and does not require gas content.

Reply:
We may need to try to delete the duplicated code